netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -97,24 +114,79 @@
   private Status goAwayStatus;
   private int nextStreamId;
 
-  public NettyClientHandler(BufferingHttp2ConnectionEncoder encoder, Http2Connection connection,
-                            Http2FrameReader frameReader,
-                            int flowControlWindow) {
-    this(encoder, connection, frameReader, flowControlWindow, Ticker.systemTicker());
+  static NettyClientHandler newHandler(Future<ClientTransport.Listener> listener,
```

Comment:
Instead of using the `Future`, could we just delay the `newHandler()` call in `NettyClientTransport` until `NettyClientTransport.start()` such that the listener would be available?